### PR TITLE
DropDownButton - add aria-expanded attribute for the buttons elements (T1051604)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/ui/drop_down_button.js [hunk 1]
```diff
@@ -451,12 +450,26 @@ const DropDownButton = Widget.inherit({
...
         this.option('opened', true);
-        this.setAria({
-            expanded: true,
-            owns: this._popupContentId
+        this._setAriaExpanded(true);
+    },
+
+    _setAriaExpanded(value) {
+        const $ariaElements = this._getExpandedAriaElements();
+        each($ariaElements, (i, ariaElement) => {
```
Comment (Zedwag) [inline js/ui/drop_down_button.js]: why not Array.forEach？

**Review by Zedwag — DISMISSED** [review]

File: js/ui/drop_down_button.js [hunk 2]
```diff
@@ -451,12 +449,26 @@ const DropDownButton = Widget.inherit({
...
+            this.setAria({
+                expanded: value,
+                owns: value ? this._popupContentId : undefined,
+            },
+            $(ariaElement));
         });
     },
 
+    _getExpandedAriaElements() {
+        const buttonElements = this._buttonGroup.$element().find('.dx-button').toArray();
```
Comment (ksercs) [inline js/ui/drop_down_button.js]: Maybe we can cache `$buttonElements` in `_renderButtonGroup` to improve performance?

File: js/ui/drop_down_button.js [hunk 3]
```diff
@@ -451,9 +449,17 @@ const DropDownButton = Widget.inherit({
...
     _popupShowingHandler() {
         this.option('opened', true);
-        this.setAria({
-            expanded: true,
-            owns: this._popupContentId
+        this._setAriaExpanded(true);
+    },
+
+    _setAriaExpanded(value) {
+        const ariaElements = [ ...this._buttonElements, this.$element() ];
```
Comment (ksercs) [inline js/ui/drop_down_button.js]: let's cache not only button elements, but all "aria expanded elements" to avoid excess destucturizations?

**Review by Zedwag — DISMISSED** [review]

File: js/ui/drop_down_button.js [hunk 4]
```diff
@@ -465,12 +470,18 @@ const DropDownButton = Widget.inherit({
 
         this._buttonGroup = this._createComponent($buttonGroup, ButtonGroup, this._buttonGroupOptions());
 
+        const buttonElements = this._buttonGroup.$element().find('.dx-button').toArray();
+
+        this.ariaExpandedElements = [ ...buttonElements, this.$element() ];
```
Comment (ksercs) [inline js/ui/drop_down_button.js]: it's a private cache. Let's name it with `_` prefix?

**Review by ksercs — APPROVED** [review]

**Review by Zedwag — APPROVED** [review]

**Pull request merged**